src/stpipe/cmdline.py

```diff
@@ -120,7 +120,8 @@ def set_value(subconf, key, val):
             set_value(subconf.setdefault(root, {}), rest, val)
         else:
             val, comment = config._handle_value(val)
-            subconf[root] = FromCommandLine(val)
+            if key in ['input_file', 'output_file']:  # or isinstance(val, str)
```

Comment:
Since the whole point of this seems to be to determine a working directory, and the presumption, in other parts of the Step I/O system, is that the working directory is determined by the input, methinks this logic is good, but only keying off of 'input_file', not both 'input_file' and 'output_file'. 

Thus the only parameter that will get this bizarre recast to `FromCommandLine` would be 'input_file'. That is a good thing methinks.